handlers.go

```diff
@@ -371,13 +320,13 @@ func (p *unitProcessor) addAllNeededUnits() {
 				requires = append(requires, appChangeID)
 				changeApplication = placeholder(appChangeID)
 			}
-			unitName := p.unitPlaceholder(name, i)
+			unitName := p.existing.nextUnit(name)
 			change := newAddUnitChange(AddUnitParams{
 				Application: changeApplication,
 				unitName:    unitName,
 			}, requires...)
 			p.add(change)
-			p.addUnitChanges[unitName] = change
+			p.addUnitChanges[p.unitPlaceholder(name, i)] = change
```

Comment:
I'm surprised you're still using the placeholder function - doesn't it tack a $ on things?